src/core_test/mpas_test_openacc.F

```diff
@@ -0,0 +1,362 @@
+    !
+    !-----------------------------------------------------------------------
+    function mpas_test_openacc(domain) result(ierr_count)
+
+        use mpas_derived_types, only : domain_type
+        use mpas_kind_types, only : StrKIND
+        use mpas_dmpar, only : mpas_dmpar_max_int
+
+        implicit none
+
+        ! Arguments
+        type (domain_type), intent(inout) :: domain
+
+        ! Return value
+        integer :: ierr_count
+
+        ! Local variables
+        integer :: ierr, ierr_global
+        ! Use routine_name to track what is being tested next
+        character(len=StrKIND) :: routine_name
+
+        ierr_count = 0
+
+        call mpas_log_write('--- Begin OpenACC tests')
+
+        routine_name = 'openacc_test_rep_arrs'
+        ierr = openacc_test_rep_arrs(domain)
+        if (ierr == 0) then
+            call mpas_log_write('   '//trim(routine_name)//' - PASSED')
+        else
+            ierr_count = ierr_count + 1
+            call mpas_log_write('   '//trim(routine_name)//' - FAILED')
+        end if
+
+        ! Deallocate any dynamic vars here
+
+        ! Make sure all threads have the max number of tests failed in
+        call mpas_dmpar_max_int(domain % dminfo, ierr, ierr_global)
+        ierr = ierr_global
+
+    end function mpas_test_openacc
+
+
+    !-----------------------------------------------------------------------
+    !  routine openacc_test_rep_arrs
+    !
+    !> \brief OpenACC test of representative of array usage
+    !> \author G. Dylan Dickerson
+    !> \date   29 May 2024
+    !> \details
+    !>  Replicates patterns from the core_atmosphere dynamics and
+    !>  compares the results on the CPU to those on the GPU. These
+    !>  patterns include a main routine that fetches arrays and
+    !>  dimensions that are passed to work routines and loops
+    !>  in the work routine that calculate some helper values before the
+    !>  result.
+    !>
+    !>  Return value: 0 (success) if the CPU and GPU results match on
+    !>  all ranks, 1 otherwise
+    !-----------------------------------------------------------------------
+    function openacc_test_rep_arrs(domain) result(ierr)
+
+        use mpas_derived_types, only : domain_type, mpas_pool_type
+        use mpas_kind_types, only : RKIND
+        use mpas_pool_routines, only : mpas_pool_get_subpool,mpas_pool_get_dimension, &
+                                       mpas_pool_get_array
+        use mpas_dmpar, only : mpas_dmpar_max_int
+
+        implicit none
+
+        ! Arguments
+        type (domain_type), intent(inout) :: domain
+
+        ! Return value
+        integer :: ierr
+
+        ! Local variables
+        integer :: ierr_global
+        real (kind=RKIND) :: diff
+        type (mpas_pool_type), pointer :: mesh_pool
+        integer, pointer :: nCells,nCellsSolve,cellSolveStart,cellSolveEnd
+        integer, pointer :: nEdges,nEdgesSolve,edgeSolveStart,edgeSolveEnd
```

Comment:
The `cellSolveStart`, `cellSolveEnd`, `edgeSolveStart`, and `edgeSolveEnd` variables are not used and could be removed.

Reply:
Removed in force push from [690909e ](https://github.com/MPAS-Dev/MPAS-Model/commit/690909ed0fb06b3037e997ed7828151e97f0ae2a)to [10fb55b ](https://github.com/MPAS-Dev/MPAS-Model/commit/10fb55bca892dec069413c0dca378ef2b21fcf12)